crates/sncast/src/starknet_commands/declare.rs

```diff
@@ -75,8 +77,14 @@ pub async fn declare(
 
     let contract_definition: SierraClass = serde_json::from_str(&contract_artifacts.sierra)
         .context("Failed to parse sierra artifact")?;
-    let casm_contract_definition: CompiledClass =
-        serde_json::from_str(&contract_artifacts.casm).context("Failed to parse casm artifact")?;
+    let casm_contract_definition: CompiledClass = if contract_artifacts.casm.is_empty() {
```

Comment:
That is cool! do you mind giving me some more context on why we should do this/why we can do it now and we couldn't earlier?

Reply:
Before we relied on Scarb to do the compilation to casm, now we have a package (will be a binary in a near future) to do sierra->casm compilation

Reply:
Yeah, we discussed with Marcin offline, but thanks!